cms/src/main/java/uk/nhs/digital/ps/modules/searchableFlagModule/PublicationProcessor.java

```diff
@@ -0,0 +1,57 @@
+package uk.nhs.digital.ps.modules.searchableFlagModule;
+
+import java.util.List;
+import javax.jcr.Node;
+import javax.jcr.NodeIterator;
+import javax.jcr.RepositoryException;
+import javax.jcr.query.Query;
+import javax.jcr.query.QueryResult;
+
+public class PublicationProcessor extends AbstractProcessor {
+
+    @Override
+    public void processNode(Node node, List<String> validStatuses) {
+        streamChildNodes(node)
+            .filter((document) -> hasValidStateAndType(document, validStatuses,"publicationsystem:publication"))
+            .forEach(this::setSearchableFlag);
+
+        // now the tricky bit - update all datasets documents belonging to this publication,
+        // only if publication name is "content"
+        try {
+            if ("content".equals(node.getName())) {
+
+                boolean searchable = node.getNode(node.getName()).getProperty("publicationsystem:PubliclyAccessible").getBoolean();
+
+                String query = "SELECT * FROM [publicationsystem:dataset] WHERE ISDESCENDANTNODE (["
+                    + node.getParent().getPath() + "])";
+
+                Query qu = node.getSession().getWorkspace().getQueryManager().createQuery(query, Query.JCR_SQL2);
+                QueryResult res = qu.execute();
+
+                for (NodeIterator i = res.getNodes(); i.hasNext(); ) {
+                    Node doc = i.nextNode();
+                    if (hasValidStateAndType(doc, validStatuses, "publicationsystem:dataset")) {
+                        updateDataset(doc, searchable);
+                    }
+                }
+            }
+        } catch (RepositoryException ex) {
+            ex.printStackTrace();
+        }
+    }
+
+    private void setSearchableFlag(Node documet) {
+        try {
+            documet.setProperty(
+                SEARCHABLE_FLAG,
+                documet.getProperty("publicationsystem:PubliclyAccessible").getBoolean()
+            );
+        } catch (RepositoryException ex) {
+            ex.printStackTrace();
+        }
+    }
+
+    private void updateDataset(Node node, boolean searchable) throws RepositoryException {
```

Comment:
Could combine this with `setSearchableFlag()` as they do the same thing with a different value for `searchable`?